runtime/stdlib.js

```diff
@@ -358,90 +358,205 @@ function caml_floatarray_create(len){
+        };
+        break;
+      case 253: // Double_tag
+        // Cannot happen
+        caml_invalid_argument("equal: got Double_tag, should not happen");
+        break;
+      case 254: // Double_array_tag
+        // Cannot happen, handled above
+        caml_invalid_argument("equal: got Double_array_tag, should not happen");
+        break
+      case 255: // Int64
+        // Int64 is the only custom block implemented this way,
+        // TODO: We should rewrite the implementation and follow
+        // what we do for other custom blocks: zarith, bigint, bigarray, ...
+        var x = caml_int64_compare(a, b);
+        if (x != 0) return (x | 0);
+        break;
+      case 1247: // Function
+        caml_invalid_argument("compare: functional value");
+        break;
+      case 1255: // Custom
+        var comp = caml_compare_val_get_custom(a);
+        if(comp != caml_compare_val_get_custom(b)){
+          return (a.caml_custom<b.caml_custom)?-1:1;
+        }
+        if(!comp)
+          caml_invalid_argument("compare: abstract value");
+        var x = comp(a,b,total);
+        if(x != x){ // Protect against invalid UNORDERED
+          return total?-1:x;
+        }
+        if(x !== (x|0)){ // Protect against invalid return value
+          return -1
+        }
+        if (x != 0) return (x | 0);
+        break;
+      case 1256: // compare function
+        var x = a.compare(b,total);
+        if(x != x) { // Protect against invalid UNORDERED
+          return total?-1:x;
+        }
+        if(x !== (x|0)){ // Protect against invalid return value
+          return -1
+        }
+        if (x != 0) return (x | 0);
+        break;
+      case 1000: // Number
+        a = +a;
+        b = +b;
         if (a < b) return -1;
         if (a > b) return 1;
         if (a != b) {
           if (!total) return NaN;
           if (a == a) return 1;
           if (b == b) return -1;
         }
+        break;
+      case 1001: // The rest
+        // Here we can be in the following cases:
+        // 1. JavaScript primitive types
+        // 2. JavaScript object that can be coerced to primitive types
+        // 3. JavaScript object than cannot be coerced to primitive types
+        //
+        // (3) will raise a [TypeError]
+        // (2) will coerce to primitive types using [valueOf] or [toString]
+        // (2) and (3), after eventual coercion
+        // - if a and b are strings, apply lexicographic comparison
+        // - if a or b are not strings, convert a and b to number
+        //   and apply standard comparison
+        //
+        // Exception: `!=` will not coerce/convert if both a and b are objects
+        if (a < b) return -1;
+        if (a > b) return 1;
+        if (a != b) {
+          if (!total) return NaN;
+          if (a == a) return 1;
+          if (b == b) return -1;
+        }
+        break;
+      case 246: // Lazy_tag
+      case 1251: // JavaScript Symbol, not ordering.
+        if(a != b) {
```

Comment:
should this be `===`?

Reply:
or `!==` rather

Reply:
`!==` and `!=` would behave the same for `Symbol`.
But you're right, `!==` convey the intent better